peps/pep-0771.rst

```diff
@@ -0,0 +1,505 @@
+override these defaults would provide a much better experience for users, and
+the approach described in this PEP will allow this.
+
+Note that this PEP does not aim to address the issue of disallowing conflicting
+or incompatible extras - for example if a package requires exactly one frontend
+or backend package. There is currently no mechanism in Python packaging
+infrastructure to disallow conflicting or incompatible extras to be installed,
+and this PEP does not change that.
+
+Rationale
+=========
+
+A number of possible solutions have been extensively and vigorously discussed by
+the community for several years, including in `this DPO threadq
+<https://discuss.python.org/t/adding-a-default-extra-require-environment/4898/38>`__
+as well as in numerous issues and pull requests. The solution that will be
+presented below:
+
+* does not break backward-compatibility of existing packaging infrastructure
+* is an opt-in solution which means that package maintainers can choose whether or not to use it
+* is flexible enough to accommodate both of the major use cases described in `Motivation`_.
+
+It is the only solution out of all those discussed that meets all three criteria.
+
+Specification
+=============
+
+``Default-Extra`` Metadata Field
+---------------------------------
+
+A new metadata field, ``Default-Extra``, will be added to the `core package
+metadata <https://packaging.python.org/en/latest/specifications/core-metadata/#core-metadata>`_.
+This field allows package maintainers to define an extra that is
+automatically included when a user installs the package without specifying any
+extras explicitly::
+
+    Default-Extra: recommended
+
+If multiple default extras are needed, one ``Default-Extra:`` entry
+must be provided for each one::
+
+    Default-Extra: backend1
+    Default-Extra: backend2
+    Default-Extra: backend3
+
+New key in ``[project]`` metadata table
+---------------------------------------
+
+A new key will be added to the ``[project]`` table in project metadata as
+originally defined in :pep:`621` and now defined in the `PyPA specifications
+<https://packaging.python.org/en/latest/specifications/>`_. This key will be named
+``default-optional-dependencies`` with the following description:
+
+* `TOML <https://toml.io/>`_ type: Array of strings
+* Corresponding core metadata field: ``Default-Extra``
+
+Each string in ``default-optional-dependencies`` must be the name of an extra
+defined in `optional-dependencies
+<https://packaging.python.org/en/latest/specifications/pyproject-toml/#dependencies-optional-dependencies>`_,
+and each extra in this array will be converted to a matching ``Default-Extra``
+entry in the core package metadata. Examples of valid usage which would
+produce the example ``Default-Extra`` entries presented in the previous section are:
+
+.. code-block:: toml
+
+    [project]
+    default-optional-dependencies = [
+        "recommended"
```

Comment:
This is very much a style nit you can ignore, but here and the examples below, shall we encourage the trailing comma? Lots of good reasons in https://www.pythonmorsels.com/trailing-commas-in-python/ such as making diffs smaller when you later add another one, and it makes it easy to re-order etc:

```suggestion
        "recommended",
```